types/hapi/definitions/server/server-events.d.ts

```diff
@@ -112,7 +156,14 @@ export interface ServerEvents extends Podium {
      * @return Return value: none.
      * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-servereventsoncecriteria-listener)
      */
-    once(criteria: string | ServerEventsApplicationObject | ServerEventCriteria, listener: Function): void;
+    once(criteria: "log" | ServerEventCriteria<"log">, listener: LogEventHandler): void;
+    once(criteria: "request" | ServerEventCriteria<"request">, listener: RequestEventHandler): void;
+    once(criteria: "response" | ServerEventCriteria<"response">, listener: ResponseEventHandler): void;
+    once(criteria: "route" | ServerEventCriteria<"route">, listener: RouteEventHandler): void;
+    once(criteria: "start" | ServerEventCriteria<"start">, listener: StartEventHandler): void;
+    once(criteria: "stop" | ServerEventCriteria<"stop">, listener: StopEventHandler): void;
+    once(criteria: string | ServerEventCriteria<string>, listener: Function): void;
```

Comment:
I don't think we need the generic version, wouldn't this still permit typos to be uncaught?